src/logic/queryParams.js

```diff
@@ -35,7 +36,22 @@ import {
  * @return {string}
  */
 const getSearchQueryFromQueryParameterOrLocalStorage = route => {
-  const { [CommonQueryParameters.SearchFilters]: sq } = route?.query;
+  const {
+    [CommonQueryParameters.SearchFilters]: sq,
+    [CommonQueryParameters.PlainSearchFilters]: f,
+  } = route?.query;
+
+  // parse json then serialize
+  if (typeof f === 'string') {
```

Comment:
This will fail if there are two `f` query parameters in the URL because f will be an array. See below how `sq` type checking is handled.

Reply:
Also, processing `f` before `sq` gives `f` higher priority. So in a scenario when both parameters are available `f` will be picked instead of `sq`.

Reply:
yep that's the idea as `f` contains data that pre-exist the adoption of `sq`

Reply:
Can we add this idea as a commentary to the code so that we are not confused looking at it in a couple of months?

Reply:
ok! 👍